### PR TITLE
BUGFIX: Fix and make `nodeTypes:show` command usable

### PR DESCRIPTION
**Pull request by mhsdesign**
Fixes handling of invalid nodetype or invalid path.

And introduces new option `level` to clamp the input (previously this command was useless for bigger nodetypes):

```
flow nodeTypes:show Neos.Demo:Document.Homepage --path properties --level 1
NodeType configuration "Neos.Demo:Document.Homepage.properties":

_removed: ...
_creationDateTime: ...
_lastModificationDateTime: ...
_lastPublicationDateTime: ...
_path: ...
_name: ...
_nodeType: ...
_hidden: ...
_hiddenBeforeDateTime: ...
_hiddenAfterDateTime: ...
titleOverride: ...
metaDescription: ...
...
```

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

**Comment by mhsdesign** [comment]
the 9.0 upmerged code should be (tested)

```php
<?php

namespace Neos\ContentRepositoryRegistry\Command;

/*
 * This file is part of the Neos.ContentRepository package.
 *
 * (c) Contributors of the Neos Project - www.neos.io
 *
 * This package is Open Source Software. For the full copyright and license
 * information, please view the LICENSE file which was distributed with this
 * source code.
 */

use Neos\ContentRepository\Core\Factory\ContentRepositoryId;
use Neos\ContentRepositoryRegistry\ContentRepositoryRegistry;
use Neos\Flow\Annotations as Flow;
use Neos\Flow\Cli\CommandController;
use Symfony\Component\Yaml\Yaml;

#[Flow\Scope("singleton")]
class NodeTypesCommandController extends CommandController
{
    public function __construct(
        private readonly ContentRepositoryRegistry $contentRepositoryRegistry
    ) {
        parent::__construct();
    }

    /**
     * Shows the merged configuration (including supertypes) of a NodeType
     *
     * @param string $nodeTypeName The name of the NodeType to show
     * @param ?string $path Path of the NodeType-configuration which will be shown
     * @param ?int $level Truncate the configuration at this depth and show '...' (Usefully for only seeing the keys of the properties)
     * @param string $contentRepository Identifier of the Content Repository to determine the set of NodeTypes
     */
    public function showCommand(string $nodeTypeName, ?string $path = null, ?int $level = 0, string $contentRepository = 'default'): void
    {
        $contentRepositoryId = ContentRepositoryId::fromString($contentRepository);
        $nodeTypeManager = $this->contentRepositoryRegistry->get($contentRepositoryId)->getNodeTypeManager();

        if (!$nodeTypeManager->hasNodeType($nodeTypeName)) {
            $this->outputLine('<error>NodeType "%s" was not found!</error>', [$nodeTypeName]);
            $this->quit();
        }

        $nodeType = $nodeTypeManager->getNodeType($nodeTypeName);

        if ($path && !$nodeType->hasConfiguration($path)) {
            $this->outputLine('<b>NodeType "%s" does not have configuration "%s".</b>', [$nodeTypeName, $path]);
            $this->quit();
        }

        $configuration = $path
            ? self::truncateArrayAtLevel($nodeType->getConfiguration($path), $level)
            : [$nodeTypeName => self::truncateArrayAtLevel($nodeType->getFullConfiguration(), $level)];

        $yaml = Yaml::dump($configuration, 99);

        $this->outputLine('<b>NodeType configuration "%s":</b>', [$nodeTypeName . ($path ? ("." . $path) : "")]);
        $this->outputLine();
        $this->outputLine($yaml);
        $this->outputLine();
    }

    /**
     * Lists all declared NodeTypes grouped by namespace
     *
     * @param string|null $filter Only NodeType-names containing this string will be listed
     * @param bool $includeAbstract List abstract NodeTypes
     * @param string $contentRepository Identifier of the Content Repository to determine the set of NodeTypes
     */
    public function listCommand(?string $filter = null, bool $includeAbstract = true, string $contentRepository = 'default'): void
    {
        $contentRepositoryId = ContentRepositoryId::fromString($contentRepository);
        $nodeTypeManager = $this->contentRepositoryRegistry->get($contentRepositoryId)->getNodeTypeManager();

        $nodeTypesFound = 0;
        $nodeTypeNameSpacesWithNodeTypeNames = [];
        foreach ($nodeTypeManager->getNodeTypes($includeAbstract) as $nodeType) {
            $nodeTypeName = $nodeType->name->value;
            if (!$filter || str_contains($nodeTypeName, $filter)) {
                [$nameSpace] = explode(":", $nodeTypeName, 2);
                $nodeTypeNameSpacesWithNodeTypeNames[$nameSpace][] = $nodeTypeName;
                $nodeTypesFound++;
            }
        }

        $this->outputLine("<b>Found $nodeTypesFound NodeTypes</b>");

        foreach ($nodeTypeNameSpacesWithNodeTypeNames as $nameSpace => $nodeTypeNames) {
            $this->outputLine();
            $this->outputLine("<b>$nameSpace</b>");

            foreach ($nodeTypeNames as $nodeTypeName) {
                $this->output->outputFormatted($nodeTypeName, [], 2);
            }
        }
    }

    /**
     * @param int $truncateLevel 0 for no truncation and 1 to only show the first keys of the array
     * @param int $currentLevel 1 for the start and will be incremented recursively
     */
    private static function truncateArrayAtLevel(array $array, int $truncateLevel, int $currentLevel = 1): array
    {
        if ($truncateLevel <= 0) {
            return $array;
        }
        $truncatedArray = [];
        foreach ($array as $key => $value) {
            if ($currentLevel >= $truncateLevel) {
                $truncatedArray[$key] = '...'; // truncated
                continue;
            }
            if (!is_array($value)) {
                $truncatedArray[$key] = $value;
                continue;
            }
            $truncatedArray[$key] = self::truncateArrayAtLevel($value, $truncateLevel, $currentLevel + 1);
        }
        return $truncatedArray;
    }
}
```

**Review by nezaniel — APPROVED** [review]
Fine by me

File: Neos.ContentRepository/Classes/Command/NodeTypesCommandController.php [hunk 1]
```diff
@@ -27,22 +27,30 @@ class NodeTypesCommandController extends CommandController
      * Shows the merged configuration (including supertypes) of a NodeType
      *
      * @param string $nodeTypeName The name of the NodeType to show
+     * @param ?int $level Truncate the configuration at this depth and show '...' (Usefully for only seeing the keys of the properties)
      * @param ?string $path Path of the NodeType-configuration which will be shown
      */
-    public function showCommand(string $nodeTypeName, ?string $path = null): void
+    public function showCommand(string $nodeTypeName, ?string $path = null, ?int $level = 0): void
```
Comment (kdambekalns) [inline Neos.ContentRepository/Classes/Command/NodeTypesCommandController.php]:
```suggestion
    public function showCommand(string $nodeTypeName, string $path = '', int $level = 0): void
```

File: Neos.ContentRepository/Classes/Command/NodeTypesCommandController.php [hunk 2]
```diff
@@ -27,22 +27,30 @@ class NodeTypesCommandController extends CommandController
      * Shows the merged configuration (including supertypes) of a NodeType
      *
      * @param string $nodeTypeName The name of the NodeType to show
+     * @param ?int $level Truncate the configuration at this depth and show '...' (Usefully for only seeing the keys of the properties)
      * @param ?string $path Path of the NodeType-configuration which will be shown
```
Comment (ahaeslich) [inline Neos.ContentRepository/Classes/Command/NodeTypesCommandController.php]:
```suggestion
     * @param ?string $path Path of the NodeType-configuration which will be shown
     * @param ?int $level Truncate the configuration at this depth and show '...' (Usefully for only seeing the keys of the properties)
```

Comment (mhsdesign) [inline Neos.ContentRepository/Classes/Command/NodeTypesCommandController.php]:
```suggestion
     * @param string $path Path of the NodeType-configuration which will be shown
     * @param int $level Truncate the configuration at this depth and show '...' (Usefully for only seeing the keys of the properties)
```

**Review by ahaeslich — APPROVED** [review]

**Pull request merged**